packages/amplify_datastore/lib/method_channel_datastore.dart

```diff
@@ -23,18 +23,22 @@ const MethodChannel _channel = MethodChannel('com.amazonaws.amplify/datastore');
 
 /// An implementation of [AmplifyDataStore] that uses method channels.
 class AmplifyDataStoreMethodChannel extends AmplifyDataStore {
-  var _allModelsStreamFromMethodChannel = null;
+  dynamic _allModelsStreamFromMethodChannel = null;
+
+  /// Internal use constructor
+  AmplifyDataStoreMethodChannel() : super.protected();
 
   /// This method adds model schemas which is necessary to instantiate native plugins
   /// This is needed before the Amplify.configure() can be called, since the native
   /// plugins are needed to be added before that.
   Future<void> configureModelProvider(
-      {ModelProviderInterface modelProvider}) async {
+      {ModelProviderInterface? modelProvider}) async {
     try {
       return await _channel
           .invokeMethod('configureModelProvider', <String, dynamic>{
-        'modelSchemas':
-            modelProvider.modelSchemas.map((schema) => schema.toMap()).toList(),
+        'modelSchemas': modelProvider!.modelSchemas

```

Comment:
Should we be casting to non-null here when the param is nullable?

Reply:
This is a bit weird.  See comments: 

/// NOTE: modelProvider is nullable because the parent AmplifyDataStore class
  /// provides a default value

Since they extend from each other I can't make one required.. even though the implementation guarantees that it woudn't be null in AmplifyDataStore: 

```
@override
  Future<void> configureModelProvider(
      {ModelProviderInterface? modelProvider}) async {
    ModelProviderInterface provider =
        modelProvider == null ? this.modelProvider : modelProvider;
    if (provider.modelSchemas.isEmpty) {
```